### PR TITLE
start supporting some multi-telescope stuff.

### PR DESCRIPTION
**Pull request by mackaiver**
When splitting data into test and training sets this allows for keeping together rows in the tables that belong to the same array-wide event. This happens only when providing the `event_id_key` option so no changes to the old process.  

The `train_separation_model.py` has mostly been refactored but now also writes out the telescope name/type to the cross validation predictions. This is usefull for creating per telescope type performance plots.

### CODE REVIEW

**Comment by mackaiver** [comment]
Bump.

File: klaas/scripts/split_data.py [hunk 1]
```diff
@@ -31,6 +31,14 @@
     help='Name for the hdf5 group in the output',
     default='events'
 )
+@click.option(
+    '--event_id_key',
+    '-id',
```
Comment (MaxNoe) [inline klaas/scripts/split_data.py]: single dash options should be single letter, I guess.

File: klaas/scripts/train_separation_model.py [hunk 2]
```diff
@@ -14,6 +14,52 @@
 from ..feature_generation import feature_generation
 
 
+logging.basicConfig()
+log = logging.getLogger()
+
+
+def read_and_sample_data(path, key, columns_to_read, n_sample):
```
Comment (MaxNoe) [inline klaas/scripts/train_separation_model.py]: Aren't those functions the same for each `train_script`? So they probably should go in a central place, like `klaas.io`

File: klaas/scripts/train_separation_model.py [hunk 3]
```diff
@@ -14,6 +14,52 @@
...
+                'number of sampled events {} must be smaller than number events in file {} ({})'
+                .format(n_sample, path, len(df))
+            )
+            raise ValueError
+        log.info('Randomly sample {} events'.format(n_sample))
+        df = df.sample(n_sample)
+    return df
+
+
+def read_config(configuration_path):
```
Comment (MaxNoe) [inline klaas/scripts/train_separation_model.py]: Same as above

Comment (mackaiver) [reply klaas/scripts/train_separation_model.py]: its kind of specific. I guess we can think about it in a separate PR

**Comment by mackaiver** [comment]
> No changes needed in the other train_ and apply_ scripts?

Maybe.  Would prefer a separate PR for that as well.

**Comment by mackaiver** [comment]
Also there was a bug when splitting the data. I've written additional unit tests.

**Comment by MaxNoe** [comment]
Bump version please

**Pull request merged**